java/pypowsybl/src/main/java/com/powsybl/dataframe/network/extensions/DiscreteMeasurementsDataframeAdder.java

```diff
@@ -0,0 +1,116 @@
+            SeriesMetadata.stringIndex(DiscreteMeasurementsDataframeProvider.ELEMENT_ID),
+            SeriesMetadata.strings(DiscreteMeasurementsDataframeProvider.ID),
+            SeriesMetadata.strings(DiscreteMeasurementsDataframeProvider.TYPE),
+            SeriesMetadata.strings(DiscreteMeasurementsDataframeProvider.TAP_CHANGER),
+            SeriesMetadata.strings(DiscreteMeasurementsDataframeProvider.VALUE_TYPE),
+            SeriesMetadata.strings(DiscreteMeasurementsDataframeProvider.VALUE),
+            SeriesMetadata.booleans(DiscreteMeasurementsDataframeProvider.VALID)
+    );
+
+    private static class DiscreteMeasurementsSeries {
+        private final StringSeries ids;
+        private final StringSeries elementIds;
+        private final StringSeries types;
+        private final StringSeries tapChangers;
+        private final StringSeries valueTypes;
+        private final StringSeries values;
+        private final IntSeries valids;
+        private final Map<String, DiscreteMeasurements<?>> measurementsByElementId = new HashMap<>();
+
+        DiscreteMeasurementsSeries(UpdatingDataframe dataframe) {
+            this.elementIds = dataframe.getStrings(DiscreteMeasurementsDataframeProvider.ELEMENT_ID);
+            this.ids = dataframe.getStrings(DiscreteMeasurementsDataframeProvider.ID);
+            this.types = dataframe.getStrings(DiscreteMeasurementsDataframeProvider.TYPE);
+            this.tapChangers = dataframe.getStrings(DiscreteMeasurementsDataframeProvider.TAP_CHANGER);
+            this.valueTypes = dataframe.getStrings(DiscreteMeasurementsDataframeProvider.VALUE_TYPE);
+            this.values = dataframe.getStrings(DiscreteMeasurementsDataframeProvider.VALUE);
+            this.valids = dataframe.getInts(DiscreteMeasurementsDataframeProvider.VALID);
+        }
+
+        void create(int row) {
+            String elementId = this.elementIds.get(row);
+            DiscreteMeasurements<?> measurements = measurementsByElementId.get(elementId);
+            DiscreteMeasurementAdder adder = measurements.newDiscreteMeasurement();
+            SeriesUtils.applyIfPresent(ids, row, adder::setId);
+            SeriesUtils.applyIfPresent(types, row, type -> adder.setType(DiscreteMeasurement.Type.valueOf(type)));
+            SeriesUtils.applyIfPresent(tapChangers, row, tapChanger -> adder.setTapChanger(DiscreteMeasurement.TapChanger.valueOf(tapChanger)));
+            SeriesUtils.applyIfPresent(values, row, value -> {
+                switch (DiscreteMeasurement.ValueType.valueOf(valueTypes.get(row))) {
+                    case BOOLEAN -> adder.setValue(Boolean.parseBoolean(value));
+                    case INT -> adder.setValue(Integer.parseInt(value));
+                    case STRING -> adder.setValue(value);
+                }
+            });
+            SeriesUtils.applyBooleanIfPresent(valids, row, adder::setValid);
+            adder.add();
+        }
+
+        @SuppressWarnings("unchecked")
```

Comment:
you don't need the annotation anymore, do you?

Reply:
Needed